src/pkg/setup/setup.go

```diff
@@ -0,0 +1,267 @@
+	beforeGenerate(folder)
+	term.Info("Working on it. This may take 1 or 2 minutes...")
+	args := cli.GenerateArgs{
+		Description: prompt.Description,
+		Folder:      folder,
+		Language:    prompt.Language,
+		ModelId:     prompt.ModelID,
+	}
+	_, err = cli.GenerateWithAI(ctx, s.Fabric, args)
+	if err != nil {
+		return SetupResult{}, err
+	}
+	return SetupResult{Folder: folder}, nil
+}
+
+type SetupResult struct {
+	Folder string
+}
+
+func (s *SetupClient) CloneSample(ctx context.Context, sample string) (SetupResult, error) {
+	var err error
+	if sample == "" {
+		sample, err = promptForSample(ctx)
+		if err != nil {
+			return SetupResult{}, fmt.Errorf("failed to prompt for sample: %w", err)
+		}
+	}
+	if sample == generateWithAI {
+		return s.AIGenerate(ctx)
+	}
+
+	folder, err := promptForDirectory(sample)
+	if err != nil {
+		return SetupResult{}, err
+	}
+	track.Evt(GenerateStartedEvt, P("sample", sample), P("folder", folder))
+	beforeGenerate(folder)
+	term.Info("Fetching sample from the Defang repository...")
+	err = cli.InitFromSamples(ctx, folder, []string{sample})
+	if err != nil {
+		return SetupResult{}, err
+	}
+
+	return SetupResult{Folder: folder}, nil
+}
+
+func promptForDirectory(defaultDirectory string) (string, error) {
+	var folder string
+	err := survey.AskOne(&survey.Input{
+		Message: "What folder would you like to create the project in?",
+		Default: defaultDirectory,
+		Help:    "The generated code will be in the folder you choose here. If the folder does not exist, it will be created.",
+	}, &folder, survey.WithStdio(term.DefaultTerm.Stdio()))
+	if err != nil {
+		return "", err
+	}
+
+	return strings.TrimSpace(folder), nil
+}
+
+type GeneratePrompt struct {
+	Description string `json:"description"`
+	ModelID     string `json:"model_id"`
+	Language    string `json:"language"`
+}
+
+func beforeGenerate(directory string) {
+	// Check if the current folder is empty
+	if empty, err := pkg.IsDirEmpty(directory); !os.IsNotExist(err) && !empty {
+		nonEmptyFolder := fmt.Sprintf("The folder %q is not empty. We recommend running this command in an empty folder.", directory)
+
+		var confirm bool
+		err := survey.AskOne(&survey.Confirm{
+			Message: nonEmptyFolder + " Continue creating project?",
+		}, &confirm, survey.WithStdio(term.DefaultTerm.Stdio()))
+		if err == nil && !confirm {
+			os.Exit(1)
+		}
+	}
+}
+
+func (s *SetupClient) MigrateFromHeroku(ctx context.Context) (SetupResult, error) {
+	var composeFileContents string
+
+	term.Info("Ok, let's create a compose file for your existing deployment.")
+	composeFileContents, err := migrate.InteractiveSetup(ctx, s.Fabric, s.Surveyor, s.Heroku, "heroku")
+	if err != nil {
+		return SetupResult{}, err
+	}
+
+	composeFilePath, err := writeComposeFile(composeFileContents)
+	if err != nil {
+		return SetupResult{}, fmt.Errorf("failed to write compose file: %w", err)
+	}
+
+	term.Info("Compose file written to", composeFilePath)
+	return SetupResult{Folder: "."}, nil
```

Comment:
Would make sense to also include the resulting `composeFilePath` to the `SetupResult`